test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -127,13 +127,24 @@ object TestAspect extends TimeoutVariants {
   /**
    * Constructs an aspect that evaluates every test inside the context of a `Managed`.
    */
-  def around[R0, E0](before: ZIO[R0, E0, Any], after: ZIO[R0, Nothing, Any]) =
+  def around[R0, E0](
+    before: ZIO[R0, E0, Any],
+    after: ZIO[R0, Nothing, Any]
+  ): PerTest[Nothing, R0, E0, Any, Nothing, Any] = around(before)(_ => after)
+
+  /**
+   * Constructs an aspect that evaluates every test inside the context of a `Managed` where the result of `before` can
+   * be used in `after`.
+   */
+  def around[R0, E0, A0](
+    before: ZIO[R0, E0, A0]
+  )(after: A0 => ZIO[R0, Nothing, Any]): PerTest[Nothing, R0, E0, Any, Nothing, Any] =
     new TestAspect.PerTest[Nothing, R0, E0, Any, Nothing, Any] {
       def perTest[R >: Nothing <: R0, E >: E0 <: Any, S >: Nothing <: Any](
         test: ZIO[R, TestFailure[E], TestSuccess[S]]
       ): ZIO[R, TestFailure[E], TestSuccess[S]] =
         ZManaged
```

Comment:
I think we can use `ZIO#bracket` here instead of `ZManaged`. Since we're just using the resource and not composing it with other resources there is no need for the overhead of a `ZManaged`.